kafka-plugins-0.10/src/main/java/io/cdap/plugin/batch/source/KafkaBatchSource.java

```diff
@@ -167,10 +168,10 @@ public void prepareRun(BatchSourceContext context) throws Exception {
     KafkaHelpers.setupKerberosLogin(kafkaConf, config.getPrincipal(), config.getKeytabLocation());
     kafkaConf.putAll(config.getKafkaProperties());
     kafkaRequests = KafkaInputFormat.saveKafkaRequests(conf, config.getTopic(), kafkaConf,
-                                                       config.getPartitions(), config.getMaxNumberRecords(),
+                                                       config.getPartitions(context.getFailureCollector()), config.getMaxNumberRecords(),
```

Comment:
should perform all these gets at the start of the method, then call getOrThrowException on the failure collector.